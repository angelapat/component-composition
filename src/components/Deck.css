.Flex {
  display: flex;
  width: 100%;
  overflow: hidden;

  * {
    max-width: 33vw;
  }
}
