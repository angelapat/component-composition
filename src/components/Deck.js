import React from 'react';
import PropTypes from 'prop-types';
import styles from './Deck.css';

export default function Deck({ children, title, justify }) {
  return (
    <>
      {title && <h1>{title}</h1>}
      <section className={styles.Flex} style={{ justifyContent: justify }}>
        {children}
      </section>
    </>
  );
};

Deck.propTypes = {
  title: PropTypes.string,
  justify: PropTypes.string,
  children: PropTypes.node.isRequired
};

Deck.defaultProps = {
  justify: 'center'
};