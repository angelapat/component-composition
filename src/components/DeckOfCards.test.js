import React from 'react';
import { shallow } from 'enzyme';
import Deck from './DeckOfCards';

describe('Deck component', () => {

  it('renders Deck', () => {
    const characters = [
      {
        _id: "5da23754845fd2cb76d59f0b",
        image: "https://media.tenor.com/images/8f6432a3fae8f961b38c107fb46f6722/tenor.gif"
      },
      {
        _id: "5da237699734fdcb7bef8f52",
        name: "Grandpa Phil",
        image: "https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      }
    ] 
    const deck = shallow(<Deck characters={characters} />
    );
    expect(deck).toMatchSnapshot();
  });
});

