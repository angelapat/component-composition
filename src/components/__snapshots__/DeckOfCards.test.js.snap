// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOfCards component renders deck of cards 1`] = `
<Deck
  justify="center"
>
  <Card
    _id="5da23754845fd2cb76d59f0b"
    image="https://media.tenor.com/images/8f6432a3fae8f961b38c107fb46f6722/tenor.gif"
    key="5da23754845fd2cb76d59f0b"
  />
  <Card
    _id="5da237699734fdcb7bef8f52"
    image="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
    key="5da237699734fdcb7bef8f52"
    name="Grandpa Phil"
  />
</Deck>
`;
